logging/gateway_interceptor.go

```diff
@@ -0,0 +1,172 @@
+package logging
+
+import (
+	"context"
+	"path"
+	"time"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/google/uuid"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/logrus/ctxlogrus"
+	"github.com/infobloxopen/atlas-app-toolkit/auth"
+	"github.com/infobloxopen/atlas-app-toolkit/gateway"
+	"github.com/infobloxopen/atlas-app-toolkit/requestid"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type gwLogCfg struct {
+	dynamicLogLvl bool
+	noRequestID   bool
+	acctIDKeyfunc jwt.Keyfunc
+	withAcctID    bool
+}
+
+// GWLogOption is a type of function that alters a gwLogCfg in the instantiation
+// of a GatewayLoggingInterceptor
+type GWLogOption func(*gwLogCfg)
+
+// NoRequestID disables request-id inclusion (and generation if needed) in gw interceptor logs
+func NoRequestID(o *gwLogCfg) {
```

Comment:
Can this be a verb? Maybe
```suggestion
func OmitRequestID(o *gwLogCfg) {
```
or
```suggestion
func DisableRequestID(o *gwLogCfg) {
```